### ISSUE TITLE
[Bug]: Unable to install package

### ISSUE THREAD

**Issue by IAmHereToFix** [pyrevitlabs/pyRevit#2315]
### ✈ Pre-Flight checks

- [X] I **don't have SentinelOne** antivirus installed (see above for the solution)
- [X] I have **searched in the issues** (open and closed) but couldn't find a similar issue
- [X] I have **searched in the [pyRevit Forum](https://discourse.pyrevitlabs.io)** for similar issues
- [X] I already **followed the [installation troubleshooting guide](https://pyrevitlabs.notion.site/Installation-issues-f6495adb56254e9e8021f8a68e3b2ab7) thoroughly**
- [X] I am using the **latest pyRevit Version**

### 🐞 Describe the bug

Hi, 

We have a user having issues installing pyrevit with the following error.
I am for sure this has to do with our company firewall blocking access to pyrevit but just wanted to check if there are other configurations that I might be able to change to fix the issue? And is there any website links that needs to be whitelisted on our firewall to allow the package download?

![2024-07-19 12_33_27-INC0300656 _ Stantec and 3 more pages - Work - Microsoft​ Edge](https://github.com/user-attachments/assets/4e30e05d-d185-4a8f-91ac-0efdac61bb06)

![2024-07-19 12_33_41-INC0300656 _ Stantec and 3 more pages - Work - Microsoft​ Edge](https://github.com/user-attachments/assets/cbeef19f-453a-43db-a5e1-1874def5e2ec)

Thank you

### ⌨ Error/Debug Message

```shell
Have reinstalled Revit and also pyrevit
The extension loads up but we are having issues downloading the package
```

### 🖥️ Hardware and Software Setup (please complete the following information)

```shell
C:\Users\xxxxxxx>pyrevit env
==> Registered Clones (full git repos)
==> Registered Clones (deployed from archive/image)
master | Deploy: "basepublic" | Branch: "master" | Version: "4.8.16.24121+2117" | Path: "C:\Users\xxxxxxx\AppData\Roaming\pyRevit-Master"
==> Attachments
master | Product: "24.1.1" | Engine: DEFAULT (2711) | Path: "C:\Users\xxxxxxx\AppData\Roaming\pyRevit-Master"
master | Product: "23.1.3" | Engine: DEFAULT (2711) | Path: "C:\Users\xxxxxxx\AppData\Roaming\pyRevit-Master"
master | Product: "2022.1.5" | Engine: DEFAULT (2711) | Path: "C:\Users\xxxxxxx\AppData\Roaming\pyRevit-Master"
master | Product: "2021.1.7" | Engine: DEFAULT (2711) | Path: "C:\Users\xxxxxxx\AppData\Roaming\pyRevit-Master"
==> Installed Extensions
==> Default Extension Search Path
C:\Users\xxxxxxx\AppData\Roaming\pyRevit\Extensions
==> Extension Search Paths
==> Extension Sources - Default
https://github.com/pyrevitlabs/pyRevit/raw/master/extensions/extensions.json
==> Extension Sources - Additional
==> Installed Revits
24.1.1 | Version: 24.1.10.25 | Build: 20230830_0315(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2024\"
23.1.3 | Version: 23.1.30.97 | Build: 20230828_1515(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2023\"
2022.1.5 | Version: 22.1.50.17 | Build: 20230915_1530(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2022\"
2021.1.7 | Version: 21.1.70.21 | Build: 20220517_1515(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2021\"
==> Running Revit Instances
PID: 26116 | 24.1.1 | Version: 24.1.10.25 | Build: 20230830_0315(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit 2024"
==> User Environment
Microsoft Windows 10 [Version 10.0.22621]
Executing User: 
Active User: 
Admin Access: No
%APPDATA%: "C:\Users\xxxxxxx\AppData\Roaming"
Latest Installed .Net Framework: 4.8
No .Net Target Packs are installed.
No .Ne-Core Target Packs are installed.
pyRevit CLI v4.8.16.24121+2117.23a0b8cbc4babdeb29a0611e159d017bbe9cbad5
```

**Comment by SpaceArchitecture** [comment]
That might be the same problem I have

**Comment by sanzoghenzo** [comment]
Can you please try to click the button while holding the CTRL key, in order to (maybe) get a more verbose output?

Extensions are declared in this file
https://github.com/pyrevitlabs/pyRevit/blob/master/extensions/extensions.json

As you can see in the "url" attributes, it only uses github.com

**Comment by sanzoghenzo** [comment]
I took a better look at the code that checks the connection and I saw that it checks if at least one of the following urls is reachable:

https://github.com/pyrevitlabs/pyRevit/blob/f9fa0d296386d548e198324dd752fb7fb5cd7b58/pyrevitlib/pyrevit/coreutils/__init__.py#L601-L608

Could it be that the firewall is blocking all (or most of) the outgoing traffic from the Revit application/process?

**Comment by IAmHereToFix** [comment]
The links are not blocked, but we will get our network team to help troubleshoot the issue.
Most probably there is a url that is getting blocked.
Will provide an update once we manage to determine what is blocking the update.

**Comment by sanzoghenzo** [comment]
To be more clear, that list of url is shuffled and the check passes as soon as one of the url is reachable.

For the check to fail, **all** of the url must be unreachable.

In my company, I often have problems when downloading pachages from npm or other repositories because of failed SSL verifications; I still don't know exactly what's the culprit since we have many (annoying) layers of security (sentinel one, symantec wss, fortigate firewall), but I suspect something in there is being overzealous. Maybe it's your case, too

**Comment by sanzoghenzo** [comment]
Yesterday @jmcouffin merged #2432 that should solve the internet access check; 

If you still experience the "certificate revocation status could not be verified", please verify the firewall setting with your IT department.
